src/Screens/Flows/App/SendScreen/02-SelectAmountSendScreen/SelectAmountSendComponent.tsx

```diff
@@ -0,0 +1,427 @@
+
+    const tokenBalance = useMemo(() => {
+        if (!selectedToken) return ""
+        const humanBalance = BigNutils(selectedToken.balance.balance).toHuman(selectedToken.decimals ?? 0)
+        return formatFullPrecision(humanBalance.toString, {
+            locale: formatLocale,
+            tokenSymbol: selectedToken.symbol,
+        })
+    }, [formatLocale, selectedToken])
+
+    const formattedInputDisplay = useMemo(() => {
+        if (!input || input === "0") return "0"
+
+        let locale: string
+        let decimalSeparator: string
+
+        switch (currencyFormat) {
+            case CURRENCY_FORMATS.COMMA:
+                locale = "de-DE"
+                decimalSeparator = CURRENCY_FORMATS.COMMA
+                break
+            case CURRENCY_FORMATS.DOT:
+                locale = "en-US"
+                decimalSeparator = CURRENCY_FORMATS.DOT
+                break
+            case CURRENCY_FORMATS.SYSTEM:
+            default:
+                locale = formatLocale
+                decimalSeparator = getDecimalSeparator(locale) ?? CURRENCY_FORMATS.DOT
+                break
+        }
+
+        const [integerPart, decimalPart] = input.split(/[.,]/)
+
+        const formatter = getNumberFormatter({
+            locale,
+            useGrouping: true,
+            precision: 0,
+            style: "decimal",
+        })
+
+        const formattedInteger = formatter.format(Number(integerPart))
+
+        if (decimalPart !== undefined) {
+            return `${formattedInteger}${decimalSeparator}${decimalPart}`
+        }
+
+        return formattedInteger
+    }, [input, currencyFormat, formatLocale])
+
+    const availableWidth = screenWidth - (24 * 2 + 50 + 20)
+
+    const totalDisplayLength = useMemo(() => {
+        const displayLength = (formattedInputDisplay || "0").length
+        if (!isInputInFiat && selectedToken?.symbol) {
+            return displayLength + 1 + selectedToken.symbol.length
+        }
+        return displayLength
+    }, [formattedInputDisplay, isInputInFiat, selectedToken?.symbol])
+
+    const inputLength = useDerivedValue(() => {
+        return totalDisplayLength
+    }, [totalDisplayLength])
+
+    const animatedInputStyle = useAnimatedStyle(() => {
+        const length = inputLength.value
+        const baseFontSize = 48
+        const minFontSize = 24
+
+        const charWidthAtBaseSize = baseFontSize * 0.6
+
+        const threshold = Math.floor(availableWidth / charWidthAtBaseSize)
+
+        let fontSize = baseFontSize
+        if (length > threshold) {
+            const targetCharWidth = availableWidth / length
+            const calculatedFontSize = targetCharWidth / 0.6
+            fontSize = Math.max(minFontSize, calculatedFontSize)
+        }
+
+        return {
+            fontSize: withTiming(fontSize, { duration: 200 }),
+            lineHeight: withTiming(fontSize, { duration: 200 }),
+        }
+    }, [inputLength, availableWidth])
```

Comment:
Either use `adjustFontSizeToFit` on the input text itself, or use this but use the FontUtils methods to display the font size consistently between iOS and Android